cnxarchive/views.py

```diff
@@ -516,6 +516,79 @@ def get_export(request):
     return resp
 
 
+@view_config(route_name='in-book-search', request_method='GET')
+def in_book_search(request):
+    """ Full text, in-book search """
+    empty_response = json.dumps({
+        u'query': {
+            u'limits': [],
+            u'per_page': DEFAULT_PER_PAGE,
+            u'page': 1,
+            },
+        u'results': {
+            u'items': [],
+            u'total': 0,
+            u'limits': [],
+            },
+    })
+        
```

Comment:
This needs something like:
    args = request.matchdict
    ident_hash = args['ident_hash']
    id, version = split_ident_hash(ident_hash)
    if version:
        if '.' in version:
            major_version, minor_version = [int(v) for v in version.split('.')]
        else:
            major_version = int(version)
            minor_version = None

Then, down inside the cursor scope/context:
    with psycopg2.connect(connection_string) as db_connection:
        with db_connection.cursor() as cursor:
            if not version:
                redirect_to_latest(cursor, id, 'in-book-search', route_args=request.params.copy())